common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/VropsPackageStore.java

```diff
@@ -999,6 +1118,105 @@ private void importPolicies(final Package vropsPackage, final File tmpDir) {
         }
     }
 
+	/**
+	 * Set the default policy in vROPs. Note it will not do anything if:
+     * 1. vROPs version is below 8.12 (as it is not supported by the vROPS API).
+     * 2. No default policy is set in the content.yaml file.
+	 * @param vropsPackage package to read policies from.
+	 * @param tmpDir temporary directory where the content.yaml file resides.
+	 * @throws RuntimeException if the default policy cannot be set.
+	 */
+    private void setDefaultPolicy(final Package vropsPackage, final File tmpDir) {
+        if (!tmpDir.exists()) {
+            return;
+        }
+        File contentYamlFile = new File(tmpDir, CONTENT_YAML_FILE_NAME);
+        VropsPackageDescriptor descriptor = this.parseContentYamlFile(contentYamlFile);
+
+        if (descriptor.getDefaultPolicy() == null || descriptor.getDefaultPolicy().size() == 0) {
+            return;
+        }
+        String defaultPolicyName = descriptor.getDefaultPolicy().stream().findFirst().orElse(null);
+        if (StringUtils.isEmpty(defaultPolicyName)) {
+            return;
+        }
+        try {
+            restClient.setDefaultPolicy(defaultPolicyName);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+	/**
+	 * Parse the content.yaml file.
+	 * @param contentYamlFile content.yaml file handle.
+	 * @return VropsPackageDescriptor POJO with the populated values from the content.yaml file.
+	 * @throws RuntimeException if the content yaml cannot be parsed.
+	 */
+    @SuppressWarnings("deprecation")
+	private VropsPackageDescriptor parseContentYamlFile(final File contentYamlFile) {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        mapper.setPropertyNamingStrategy(PropertyNamingStrategy.KEBAB_CASE);
+        try {
+            return mapper.readValue(contentYamlFile, VropsPackageDescriptor.class);
+        } catch (Exception e) {
+            throw new RuntimeException(String.format("Unable to load vROps Package Descriptor ['%s'] : %s", contentYamlFile.getName(), e.getMessage()));
+        }
+    }
+
+	/**
+	 * Updates the default policy in the content yaml file (if there is one set).
+	 * @param packageDescriptor the package descriptor POJO that will be serialized to the content.yaml file.
+	 * @param destDir destination directory where the content.yaml file resides.
+	 */
+	private void setDefaultPolicyInContentYaml(final VropsPackageDescriptor packageDescriptor, final File destDir) {
```

Comment:
Trying to understand the logic here:
1. We will pull the DefaultPolicy regardless of what the name is as long as "a string" is set in the content.yaml -> at this point, why not make the defaultPolicy a boolean?
    1. If found, we'll also update the value in the content.yaml 
2. If it's not set in the content.yaml it will skip and do nothing

This is going sideways of what we usually do and what the vrops archetype does, where we specify what we need to download and only that gets downloaded. Perhaps, we need to rethink the way this functionality should function.. in my eyes just set it as true or false

Reply:
Actually your understanding is not quite correct, the default policy is set in the content.yaml during pull in case there is no default policy set in it and the vROPs version is above 8.12, in all other cases it would do nothing.

Reply:
The comment after that still stands, even in this situation:
"This is going sideways of what we usually do and what the vrops archetype does, where we specify what we need to download and only that gets downloaded. Perhaps, we need to rethink the way this functionality should function.. in my eyes just set it as true or false"

Reply:
Could you elaborate a bit on your comment, perhaps with some code sample.